### PR TITLE
Update deps. Use BioLink API

### PR DESCRIPTION
**Pull request by DoctorBud**
- update packages, especially d3 to 5.7.0, jquery to 3.3.1, jqueryui to 1.12.1.
- Add ability to use BioLink API instead of Legacy API.
- Add '/?legacy=1' URL parameter to index.html example to allow compare/contrast between BioLink and Legacy API.

### CODE REVIEW

**Comment by DoctorBud** [comment]
The TravisCI randomly fails, but I think that is due to the /sim/search or BioLink or even SciGraph or Owlsim. I think this PR is suitable for review and merging.

File: js/dataloader.js [hunk 1]
```diff
@@ -11,17 +11,77 @@ var Utils = require('./utils.js');
...
+		        inputItemsString: 'input_items=', // HTTP POST, body parameter
+		        targetSpeciesString: '&target_species=', // HTTP POST, body parameter
+		        limitString: '&limit'
+		    };
+		}
+	}
+	else {
+		if (useBioLink) {
+			this.simQuery = { // compare API takes HTTP GET, so no body parameters
+                URL: '/compare' // used for owlSimFunction === 'compare' and genotype expansion compare simsearch - Joe
```
Comment (DoctorBud) [inline js/dataloader.js]: Write better comment here to indicate that BioLink Compare has not been tested or utilized.

File: js/dataloader.js [hunk 2]
```diff
@@ -232,37 +312,131 @@ DataLoader.prototype = {
...
 	 		callback
 	 		postData - data to be posted
-	*/ 
-	postFetch: function (url, target, targets, callback, postData) {
+	*/
+	postFetch: function (url, queryURL, target, targets, callback, qryString, targetSpeciesString, targetGroupId) {
 		var self = this;
 
-        console.log('POST:' + url);
+	   	if (url.indexOf('https://api.monarchinitiative.org') === 0) {
```
Comment (DoctorBud) [inline js/dataloader.js]: Use  the isBiolinkServer() function here.

File: js/dataloader.js [hunk 3]
```diff
@@ -232,37 +312,131 @@ DataLoader.prototype = {
...
+
+		//
+		// Apparently, monarch-app has been returning
+		// the following hardcoded data from its endpoint, as
+		// provided by Analyze.js:
+		//  https://github.com/monarch-initiative/monarch-app/blob/master/js/Analyze.js#L440
+		//
+		// For now, we'll just add in the exact same (and likely bogus) data.
+		//
+		var legacyMetadata = {
```
Comment (DoctorBud) [inline js/dataloader.js]: Delete this and assume that maxMaxIC is available from BioLink.

File: js/dataloader.js [hunk 4]
```diff
@@ -317,6 +492,8 @@ DataLoader.prototype = {
                 this.sourceData[targetGroup] = {};
             }
 
+			var logged = true;
```
Comment (DoctorBud) [inline js/dataloader.js]: Lose the logging.

File: js/phenogrid.js [hunk 5]
```diff
@@ -20,10 +20,63 @@
...
+require('jquery-ui/ui/widgets/tabs');
+require('jquery-ui/ui/widgets/dialog');
+
+*/
+
 var d3 = require('d3');
-var filesaver = require('filesaver.js');
+global.d3 = d3;
+
+var filesaver = require('file-saver');
```
Comment (DoctorBud) [inline js/phenogrid.js]: SVG Save doesn't work right anymore. Works on production legacy.

**Comment by kshefchek** [comment]
+1

**Pull request merged**